### PR TITLE
[SPARK-34232][CORE] Redact SparkListenerEnvironmentUpdate event in log

### PR DESCRIPTION
**Pull request by warrenzhu25**
### What changes were proposed in this pull request?
Redact event SparkListenerEnvironmentUpdate in log when its processing time exceeded logSlowEventThreshold

### Why are the changes needed?
Credentials could be exposed when its processing time exceeded logSlowEventThreshold

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Manually tested

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/util/ListenerBus.scala
Comment (HeartSaVioR) [inline core/src/main/scala/org/apache/spark/util/ListenerBus.scala:157]: Let's reuse `EventLoggingListener.redactEvent`. Probably change the code in EventLoggingListener like below?

```
private[spark] class EventLoggingListener(

...

  private[spark] def redactEvent(
      event: SparkListenerEnvironmentUpdate): SparkListenerEnvironmentUpdate = {
    EventLoggingListener.redactEvent(sparkConf, event)
  }
}

private[spark] object EventLoggingListener extends Logging {

...

  def redactEvent(
      sparkConf: SparkConf,
      event: SparkListenerEnvironmentUpdate): SparkListenerEnvironmentUpdate = {
    // environmentDetails maps a string descriptor to a set of properties
    // Similar to:
    // "JVM Information" -> jvmInformation,
    // "Spark Properties" -> sparkProperties,
    // ...
    // where jvmInformation, sparkProperties, etc. are sequence of tuples.
    // We go through the various  of properties and redact sensitive information from them.
    val noRedactProps = Seq("Classpath Entries")
    val redactedProps = event.environmentDetails.map {
      case (name, props) if noRedactProps.contains(name) => name -> props
      case (name, props) => name -> Utils.redact(sparkConf, props)
    }
    SparkListenerEnvironmentUpdate(redactedProps)
  }
}
```

**Comment by HeartSaVioR** [comment]
ok to test

**Comment by HeartSaVioR** [comment]
Looks OK, but the code change is proposed by me so I'd feel comfort to wait for another review.

**Comment by HyukjinKwon** [comment]
ok to test

**Review by HyukjinKwon — APPROVED** [review]

File: core/src/main/scala/org/apache/spark/util/ListenerBus.scala
Comment (dongjoon-hyun) [inline core/src/main/scala/org/apache/spark/util/ListenerBus.scala:155]: Since this is designed to be used in `ListenerBus` only, `private` will be enough in this case. For the other classes, this PR makes `EventLoggingListener.redactEvent` available already.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (only one minor comment about function visibility).

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.1.

**Pull request closed without merging**
